backend/apid/middlewares/authorization_attributes.go

```diff
@@ -0,0 +1,59 @@
+package middlewares
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/sensu/sensu-go/backend/authorization"
+)
+
+// AuthorizationAttributes is an HTTP middleware that populates a context for the
+// request, to be used by other middlewares. You probably want this middleware
+// to be executed early in the middleware stack.
+type AuthorizationAttributes struct{}
+
+// Then is the AuthorizationAttrs middleware's main logic, heavily inspired by
+// Kubernetes' WithRequestInfo.
+//
+// The general, expected format of a path one of the following: -
+// /apis/{group}/{version}/namespaces, to create or list namespaces -
+// /apis/{group}/{version}/namespaces/{name} -
+// /apis/{group}/{version}/namespaces/{namespace}/{kind} -
+// /apis/{group}/{version}/namespaces/{namespace}/{kind}/{name}
```

Comment:
The problem here is that we need to support the _old_ routes for now, until https://github.com/sensu/sensu-go/issues/2249 is implemented (probably next sprint).

Could you try to see if we could do that without too much effort?

Reply:
Sure, I thought actually plugging the routes and middleware in would be part of another PR. What are the "old" routes supposed to look like? I'm assuming something like:
- `/rbac/rules`
- `/rbac/roles`
- ...

You've added a router for `roles` already; I can do the same with the other resources (`rules`, `clusterroles`). Am I missing anything? I'll also register those new routers in `registerRestrictedResources()` and plug in the `AuthorizationAttributes` middleware.

Reply:
Just realized the challenge here is that we will have RBAC routes use the new pattern but the rest of the objects will continue to use the older patterns. Maybe we could split that logic into 2 different middlewares and once we are fully migrated, we just remove the _old_ one?

Let's sync about the routers!